kolibri/core/kolibri_plugin.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+"""
+The core app of Kolibri also uses the plugin API <3
+"""
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from django.utils.translation import ugettext as _
+from django.core.urlresolvers import reverse
+
+
+from kolibri.plugins.base import KolibriPluginBase
+from kolibri.plugins.hooks import NAVIGATION_POPULATE
+
+
+class KolibriCore(KolibriPluginBase):
```

Comment:
I was wondering if we could implement `hooks` as an attribute instead of a callable. It would look something like this:

```python
def main_navigation():
    return [{
        'menu_name': _("Start page"),
        'menu_url': reverse('kolibri:index'),
    }]


class KolibriCore(KolibriPluginBase):
    hooks = {
        NAVIGATION_POPULATE: main_navigation
    }
```

I also wondered in what cases you would want to make it callable -- I guess if you wanted to inherit from an existing plugin class and extend a class method? Can you elaborate on your intent? But anyway you could do it like this using the `property` decorator:

```python
class ExamplePlugin(KolibriPluginBase):

    @classmethod
    def enable(cls, config):
        print("Activating example plugin")
        # Make this automatic and use __name__ ?
        config["INSTALLED_APPS"].append("kolibri.plugins.example_plugin")

    @classmethod
    def disable(cls, config):
        print("Deactivating example plugin")
        # Make this automatic and use __name__ ?
        config["INSTALLED_APPS"].remove("kolibri.plugins.example_plugin")

    @staticmethod
    def main_navigation():
        return [{
            'menu_name': _("Google"),
            'menu_url': 'http://google.com',
        }]

    @property
    def hooks(self):
        return {
            NAVIGATION_POPULATE: self.main_navigation
        }


class ExtendedExamplePlugin(ExamplePlugin):
    """
    Demonstrates plugin inheritance.
    """
    @classmethod
    def enable(cls, config):
        pass

    @classmethod
    def disable(cls, config):
        pass

    def main_navigation(self):
        return [
            {
                'menu_name': _('Disney'),
                'menu_url': 'http://disney.com',
            },
            {
                'menu_name': _('Yahoo'),
                'menu_url': 'http://yahoo.com',
            }
        ]
```

However this also demonstrates two potential shortcomings:

1. The plugin callback evaluation order is reversed:
![ss3](https://cloud.githubusercontent.com/assets/8888020/12498500/ec5230c8-c057-11e5-8c89-0014a849b29c.png)
Notice that the *second* plugin's callback is called before the *first*. In general can we find a way to guarantee the order of callbacks?
2. Defining several plugin classes in one module requires clunky `enable` & `diable` logic -- in particular, I can't inherit the base class' methods because it results in duplicate entries in INSTALLED_APPS. I'm forced to override the methods even though they don't do anything.